web-api/src/storage/postgres.rs

```diff
@@ -530,36 +518,28 @@ impl storage::Interaction for Storage {
             .map(|tag| (tag.as_str(), 0))
             .collect::<HashMap<_, _>>();
 
-        let mut updates = Vec::new();
+        let mut updates = HashMap::new();
 
         for id in updated_document_ids {
-            if let Some(document) = document_map.get_mut(id) {
-                updates.push(update_logic(InteractionUpdateContext {
+            if let Some((document, _)) = document_map.get_mut(id) {
+                let updated_coi = update_logic(InteractionUpdateContext {
                     document,
                     tag_weight_diff: &mut tag_weight_diff,
                     positive_cois: &mut interests.positive,
-                }));
+                });
+                // We might update the same coi min `interests` multiple times,
+                // if we do we only want to keep the latest update.
+                updates.insert(updated_coi.id, updated_coi);
```

Comment:
:exclamation: This was one of the places which could trigger the bug.

Reply:
not sure what you mean?

Update coi in-place updated the `Vec` of cois so it always has the updated cois.

Reply:
This is not part of my design but a left over from CoI handling in the app.

Reply:
> There is an additional problem here, the updated_coi must be fed to the next interaction.

not sure if i understand this? all cois are modified via the context as a ref mut, and the updated coi is stored in the map, so shouldn't all information be available to the next interaction without a need to pass the value?

Reply:
than everything is fine. I didn't look at what the update function was doing